vector/src/main/java/im/vector/adapters/RoomViewHolder.java

```diff
@@ -281,7 +309,61 @@ public void onClick(View v) {
             });
         }
 
-        BingRulesManager.RoomNotificationState roomNotificationState = session.getDataHandler().getBingRulesManager().getRoomNotificationState(room.getRoomId());
+        final Set<String> tags = room.getAccountData().getKeys();
+        final boolean isFavorite = tags != null && tags.contains(RoomTag.ROOM_TAG_FAVOURITE);
+        final BingRulesManager.RoomNotificationState roomNotificationState = session.getDataHandler().getBingRulesManager().getRoomNotificationState(room.getRoomId());
+        RoomMember member = room.getMember(session.getMyUserId());
+        final boolean isBannedKickedRoom = (null != member) && member.kickedOrBanned();
+
+        if (null != vPinSwipeLayout) {
+            vPinSwipeLayout.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (isFavorite) {
+                        updateTag(session, room.getRoomId(), null, null);
+                    } else {
+                        updateTag(session, room.getRoomId(), null, RoomTag.ROOM_TAG_FAVOURITE);
+                    }
+                }
+            });
+        }
+
+        if (null != vSilentSwipeLayout) {
+            vSilentSwipeLayout.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (null != vRoomNotificationMute) {
+                        if (vRoomNotificationMute.getVisibility() == View.VISIBLE) {
+                            vRoomNotificationMute.setVisibility(View.GONE);
+                        } else {
+                            vRoomNotificationMute.setVisibility(View.VISIBLE);
+                        }
+                        swipeLayout.close(true);
+                    }
+
+                    if (roomNotificationState.equals(BingRulesManager.RoomNotificationState.MUTE)) {
+                        onUpdateRoomNotificationsState(context, session, room.getRoomId(), BingRulesManager.RoomNotificationState.ALL_MESSAGES);
+                    } else {
+                        onUpdateRoomNotificationsState(context, session, room.getRoomId(), BingRulesManager.RoomNotificationState.MUTE);
+                    }
+                }
+            });
+        }
+
+        if (null != vExitSwipeLayout) {
+            vExitSwipeLayout.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    swipeLayout.close(true);
```

Comment:
Note: swipeLayout.close() is called twice here and in mRoomSuccessListener implementation.
We should decide where it is more relevant